xmtp_mls/src/groups/sync.rs

```diff
@@ -825,29 +835,41 @@ impl MlsGroup {
 
                     return Err(err);
                 }
-                Ok(Some((payload, post_commit_data))) => {
-                    let payload_slice = payload.as_slice();
+                Ok(Some(PublishIntentData {
+                    message,
+                    post_commit_action,
+                    staged_commit,
+                })) => {
+                    let payload_slice = message.as_slice();
+                    let has_staged_commit = staged_commit.is_some();
+                    provider.conn_ref().set_group_intent_published(
+                        intent.id,
+                        sha256(payload_slice),
+                        post_commit_action,
+                        staged_commit,
+                        openmls_group.epoch().as_u64() as i64,
+                    )?;
+                    log::debug!(
+                        "client [{}] set stored intent [{}] to state `published`",
+                        client.inbox_id(),
+                        intent.id
+                    );
 
                     client
                         .api_client
                         .send_group_messages(vec![payload_slice])
                         .await?;
```

Comment:
Now that we're marking the intent as published before sending it, if the publish fails, either:
- Network error returned from `send_group_messages`
- App dies before the request goes through

Do we have any way of making sure the intent gets re-sent?

Reply:
In the case of a network error we bubble the error back up to the caller and they can choose to retry.

In the case of the app dying, that's the big trade-off of this approach. It's just permanently in limbo. The difference from some of the alternative solutions is that it won't brick the group.

Reply:
Just FYI, we are not bubbling it up in the case of `send_message<ApiClient>()`. And in the case of `send_message_optimistic`, the message would have already been rendered to the UI, and the error from the publish may or may not be bubbled up to the user

Reply:
> , we are not bubbling it up in the case of send_message<ApiClient>(). 

We would be because `sync_until_last_intent_resolved` will never finish